core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/ObjectEntry.kt

```diff
@@ -0,0 +1,37 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.validtype.ObjectType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+
+class ObjectEntry<T : ValidType>(
+    private val key: TypeExpression<StringType>,
+    private val value: TypeExpression<T>,
+) : Resolvable {
+    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
+        val keyQuery = key.toDopeQuery(manager)
+        val valueQuery = value.toDopeQuery(manager)
+        return DopeQuery(
+            queryString = "${keyQuery.queryString} : ${valueQuery.queryString}",
+            parameters = keyQuery.parameters + valueQuery.parameters,
+        )
+    }
+}
+
+fun TypeExpression<StringType>.to(value: TypeExpression<out ValidType>) = ObjectEntry(this, value)
+
+class ObjectEntryField<T : ValidType>(private val objectType: TypeExpression<ObjectType>, private val key: String) : Field<T>(key, "") {
```

Comment:
I made it a `Field`, because an `EntryField` is nearly the same as a normal field, but it has The Object before it in the String.
Normal Field -> `test` // is a Key in a JsonDocument
EntryField -> `object`.`test` // is a Key in an embedded Object in a JsonDocument

Reply:
Isn't it exactly the same? Do we really need this additional class?

Reply:
No its not, because the ObjectEntryField has an ObjectType and a String as Attributes and the Field has two strings.
So what I coulddo is just add another constructor to the Field with the ObjectType.
Do you think thats better?

Reply:
made the ObjectEntryField a TypeExpression